### PR TITLE
V2 Prop Creation Flow with Token Buyer UI

### PR DESCRIPTION
**Pull request by bcjgit**
# TLDR

- Update prop creation page UI
- Update transaction addition modal to support lump sum (USDC or ETH) or function call proposal actions

# Testing Done

- Added ETH lump sum, USDC lump sum, and function call actions to props
- Validated that error states / invalid inputs fail as expected

### CODE REVIEW

File: packages/nouns-webapp/src/config.ts [hunk 1]
```diff
@@ -85,15 +86,19 @@ const app: Record<SupportedChains, AppConfig> = {
...
     lidoToken: '0xF4242f9d78DB7218Ad72Ee3aE14469DBDE8731eD',
+    payerContract: undefined,
   },
   [ChainId.Goerli]: {
     lidoToken: '0x2DD6530F136D2B56330792D46aF959D9EA62E276',
+    payerContract: undefined,
   },
   [ChainId.Mainnet]: {
     lidoToken: '0xae7ab96520DE3A18E5e111B5EaAb095312D7fE84',
+    payerContract: '0x94A63a8391b8d7d188d48994c4564f0946EbA000',
```
Comment (davidbrai) [inline packages/nouns-webapp/src/config.ts:97]: please change to the latest version: 0x89AbC50b1Ed7B9B0879433032203a0Bd21B80EB4

File: packages/nouns-webapp/src/components/ProposalActionsModal/steps/LumpSumReviewStep/index.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  } else if (state.lumpSumCurrency === SupportedCurrency.USDC) {
+    const signature = 'sendOrRegisterDebt(address,uint256)';
+    const abi = new utils.Interface(payerABI);
+    onActionAdd({
+      address: config.addresses.payerContract,
+      value: '0',
+      signature,
+      calldata: abi?._encodeParams(abi?.functions[signature]?.inputs, [
+        state.address,
+        state.amount,
```
Comment (davidbrai) [inline packages/nouns-webapp/src/components/ProposalActionsModal/steps/LumpSumReviewStep/index.tsx:30]: is the user expected to enter the USDC amount with decimals? I didn't see any decimal conversion
e.g, if the user wants to send 10,000 USDC, what should they enter in the form?

File: packages/nouns-webapp/src/components/ProposalActionsModal/steps/LumpSumReviewStep/index.tsx [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    onActionAdd({
+      address: state.address,
+      value: state.amount ? utils.parseEther(state.amount.toString()).toString() : '0',
+      signature: undefined,
+      calldata: '0x',
+    });
+  } else if (state.lumpSumCurrency === SupportedCurrency.USDC) {
+    const signature = 'sendOrRegisterDebt(address,uint256)';
+    const abi = new utils.Interface(payerABI);
+    onActionAdd({
```
Comment (davidbrai) [inline packages/nouns-webapp/src/components/ProposalActionsModal/steps/LumpSumReviewStep/index.tsx:24]: what about adding a transaction to fund TokenBuyer with ETH?
description in the spec [here](https://verbs.notion.site/Draft-Nouns-Proposal-Builder-v2-Requirements-0008cfe4b6654d6a8028c7c49c163d3a)

Comment (bcjgit) [reply packages/nouns-webapp/src/components/ProposalActionsModal/steps/LumpSumReviewStep/index.tsx:24]: Added some code here to add / remove a token buyer top up if the set of proposal actions has at least 1 USDC lump sum payment 


asked a question on discord about exact function to call but add / remove mechanism seems to work

**Pull request merged**